src/components/product/film/film.service.ts

```diff
@@ -0,0 +1,321 @@
+          active: true,
+          createdAt,
+        }),
+        node(prop, propLabel, {
+          active: true,
+          value,
+        }),
+      ],
+    ];
+  };
+
+  // helper method for defining properties
+  permission = (
+    property: string,
+    sg: string,
+    baseNode: string,
+    read: boolean,
+    edit: boolean
+  ) => {
+    const createdAt = DateTime.local();
+    return [
+      [
+        node(sg),
+        relation('out', '', 'permission', {
+          active: true,
+          createdAt,
+        }),
+        node('', 'Permission', {
+          property,
+          active: true,
+          read,
+          edit,
+        }),
+        relation('out', '', 'baseNode', {
+          active: true,
+          createdAt,
+        }),
+        node(baseNode),
+      ],
+    ];
+  };
+
+  // helper method for match properties
+  propMatch = (property: string, baseNode: string) => {
+    const perm = `canRead${upperFirst(property)}`;
+    return [
+      [
+        node('requestingUser'),
+        relation('in', '', 'member', { active: true }),
+        node('sg', 'SecurityGroup', { active: true }),
+        relation('out', '', 'permission', { active: true }),
+        node(perm, 'Permission', {
+          property,
+          active: true,
+          read: true,
+        }),
+        relation('out', '', 'baseNode', { active: true }),
+        node(baseNode),
+        relation('out', '', property, { active: true }),
+        node(property, 'Property', { active: true }),
+      ],
+    ];
+  };
+
+  async create(input: CreateFilm, session: ISession): Promise<Film> {
+    const checkFm = await this.db
+      .query()
+      .raw(
+        `
+        MATCH(film:FilmName {value: $name}) return film
+        `,
+        {
+          name: input.name,
+        }
+      )
+      .first();
+
+    if (checkFm) {
+      throw new BadRequestException(
+        'Film with that name already exists.',
+        'Duplicate'
+      );
+    }
+    const id = generate();
+    const createdAt = DateTime.local();
+    try {
+      await this.db
+        .query()
+        .match(matchSession(session, { withAclEdit: 'canCreateFilm' }))
+        .create([
+          [
+            node('newFilm', 'Film:BaseNode', {
+              active: true,
+              createdAt,
+              id,
+              owningOrgId: session.owningOrgId,
+            }),
+          ],
+          ...this.property('name', input.name, 'newFilm'),
+          ...this.property('range', input.range ? true : undefined, 'newFilm'),
+          ...this.property('rangeStart', input.range?.rangeStart, 'range'),
+          ...this.property('rangeEnd', input.range?.rangeEnd, 'range'),
+          [
+            node('adminSG', 'SecurityGroup', {
+              active: true,
+              createdAt,
+              name: input.name + ' admin',
+            }),
+            relation('out', '', 'member', { active: true, createdAt }),
+            node('requestingUser'),
+          ],
```

Comment:
Please refactor the code that you can call  the function `..this.permission` in single for `adminSG` and `readerSG`. 
So, change in helper function `permission` to be able for single call.